### DISCUSSION TITLE
Is it possible to customize the re-queue behavior after a job failure or preemption?

### DISCUSSION THREAD

**Discussion by Zhenshan-Jin** [kubernetes-sigs/kueue#2752] [Q&A] [answered]
Hi team,

I am wondering if it is possible to customize the requeue behavior like
- whether to put the job at the beginning of the end of the queue
- whether to restart or resume from where it is left when the job was pre-empted by the other jobs. 

I don't see any public documentation on this topic yet. If I miss anything, please let me know. 

Thank you!

**Comment by tenzen-y** [comment, marked as answer]
@Zhenshan-Jin Sorry for the late response.
Indeed, we provide the features with waitForPodsReady. So, you can learn how you can custom behavior when re-queuing happens.

https://kueue.sigs.k8s.io/docs/tasks/manage/setup_wait_for_pods_ready/#requeuing-strategy
